src/EditorFeatures/Core.Wpf/InlineRename/CommandHandlers/RenameCommandHandler_RenameHandler.cs

```diff
@@ -108,7 +109,8 @@ private static bool CanRename(RenameCommandArgs args)
         {
             return args.SubjectBuffer.TryGetWorkspace(out var workspace) &&
                 workspace.CanApplyChange(ApplyChangesKind.ChangeDocument) &&
-                args.SubjectBuffer.SupportsRename();
+                args.SubjectBuffer.SupportsRename() &&
+                !workspace.Services.GetService<IWorkspaceContextService>().IsInLspEditorContext();
```

Comment:
i dislike this.  why is this not part of the .SupportsRename extension? seems weird for the buffer to say it supports this and we then override.

Reply:
put another way, i highly dislike this sort of sprinkling aruond of a different concept that features should not have to care about.

Reply:
@CyrusNajmabadi this is ready for review now - I moved this to the supports rename, but in general I agree and don't really like sprinkling this around either, but I'm not sure I have a better idea.

Basically I need to disable the legacy editor entry points when running under the LSP editor for features that are provided by LSP (e.g. rename) but need to keep other editor entry points active (e.g. syntax classifications).

I was potentially considering perhaps modifying the extension methods to convert textbuffer/snapshot to documents (GetOpenDocumentInCurrentContextWithChanges) to return nothing when under LSP mode, but then I'd still need a separate method to allow some features to retrieve documents from the buffer (e.g. syntax classification)

Reply:
@CyrusNajmabadi I've gotten these new integration tests working again and am looking to get this in, so wanted to followup on this conversation.  Can discuss offline whenever you have a meeting free day :)